### ISSUE TITLE
Surefire fork fails on 3.5.3

### ISSUE THREAD

**Issue by oliemansm** [carlossg/docker-maven#73]
Our builds configured in Bitbucket Pipelines using the latest release 3.5.3 are all failing all of a sudden with the following stacktrace:
```
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /opt/atlassian/pipelines/agent/build && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /opt/atlassian/pipelines/agent/build/target/surefire/surefirebooter1173070707154269215.jar /opt/atlassian/pipelines/agent/build/target/surefire 2018-03-28T12-44-59_326-jvmRun1 surefire1185691907430284876tmp surefire_0554936172727229418tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /opt/atlassian/pipelines/agent/build && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /opt/atlassian/pipelines/agent/build/target/surefire/surefirebooter1173070707154269215.jar /opt/atlassian/pipelines/agent/build/target/surefire 2018-03-28T12-44-59_326-jvmRun1 surefire1185691907430284876tmp surefire_0554936172727229418tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:686)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:535)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:280)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:245)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1124)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:954)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:832)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:290)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:194)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[ERROR] -> [Help 1]
```
When we use the fixed version 3.5.2 instead of latest the builds succeed.

**Comment by raehalme** [comment]
Try downgrading `maven-surefire-plugin` to version 2.20.

**Comment by oliemansm** [comment]
That helps getting past the build problem, but now it fails on a subsequent step that does work on 3.5.2. I wouldn't expect having to downgrade a dependency though when upgrading the image.

We install the Google Cloud SDK and deploy the app using that. On 3.5.3 the same script fails with the following error:
```
/root/google-cloud-sdk/bin/gcloud: 113: exec: python: not found
/root/google-cloud-sdk/bin/gcloud: 113: exec: python: not found
/root/google-cloud-sdk/bin/gcloud: 113: exec: python: not found
```
One of other apps is deployed to AWS, and that automatic deployment fails as well on 3.5.3:
```
+ eb init cpi-idp -r eu-central-1 -p "Java 8"
ERROR: CommandError - Your project is using git, but git doesn't appear to be installed.
Have you added git to your PATH?
```

**Comment by raehalme** [comment]
[This issue](https://github.com/docker-library/openjdk/issues/180) may be related to your problem: it seems  `git` (and maybe others as well) were removed from the openjdk image.

**Comment by bartlaarhoven** [comment]
@raehalme do you think the related `openjdk` issue is also related to the original problem? I had the same problems as the OP yesterday but I couldn't figure out what it was. Downgrading surefire to 2.19.1 solved it for me, but it's still strange. Do you have a reason to believe that the `openjdk` issue will fix this?

**Comment by carlossg** [comment]
The python issue is different see #74

The original issue could be https://issues.apache.org/jira/browse/SUREFIRE-1422

Always use image SHAs to avoid these problems!

**Comment by carlossg** [comment]
You don't mention whether you are using alpine, debian oe slim images.
This should have been fixed with #71

**Comment by shankar109** [comment]
Hi.
Has there been any fix regarding the surefire Forkedbooter issue?
Thanks

Jinmi

**Comment by Sergey80** [comment]
same

**Comment by carlossg** [comment]
same what? this has been fixed in surefire 2.21.0
